extensions/wikia/AdEngine/js/context/slotsContext.js

```diff
@@ -0,0 +1,65 @@
+/*global define*/
+define('ext.wikia.adEngine.context.slotsContext', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.utils.adLogicZoneParams',
+	'wikia.geo',
+	'wikia.instantGlobals',
+	'wikia.log'
+], function (adContext, params, geo, instantGlobals, log) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.context.slotsContext',
+		slots = null;
+
+	function setStatus(slotName, status) {
+		slots[slotName] = !!status;
+	}
+
+	function setupSlots() {
+		var context = adContext.getContext(),
+			isHome = params.getPageType() === 'home';
+
+		slots = {};
+
+		setStatus('HOME_TOP_LEADERBOARD', isHome);
+		setStatus('HOME_TOP_RIGHT_BOXAD', isHome);
+		setStatus('PREFOOTER_MIDDLE_BOXAD', isHome);
+
+		setStatus('TOP_LEADERBOARD', !isHome);
+		setStatus('TOP_RIGHT_BOXAD', !isHome);
+
+		setStatus('INVISIBLE_HIGH_IMPACT_2', geo.isProperGeo(instantGlobals.wgAdDriverHighImpact2SlotCountries));
+		setStatus('INCONTENT_LEADERBOARD', geo.isProperGeo(instantGlobals.wgAdDriverIncontentLeaderboardSlotCountries));
+		setStatus('INCONTENT_PLAYER', geo.isProperGeo(instantGlobals.wgAdDriverIncontentPlayerSlotCountries));
+
+		setStatus('PREFOOTER_RIGHT_BOXAD', !context.opts.overridePrefootersSizes);
+
+		log(['Disabled slots:', slots], 'info', logGroup);
+	}
+
+	function isApplicable(slotName) {
```

Comment:
so if the slot is not defined in `slots` - for instance `LEFT_SKYSCRAPER_2` `isApplicable` returns `true` because there is no such key in object, right?